src/test/java/com/github/messenger4j/test/integration/DocumentationTest.java

```diff
@@ -203,4 +215,67 @@ public void echoExample() throws Exception {
         });
         // end::doc-EchoExample[]
     }
+    
+    @ToString(callSuper = true)
+    @EqualsAndHashCode(callSuper = true)
+    public static class CustomEvent extends BaseEvent {
+    	
+    	@Getter
+    	@Accessors(fluent=true)
+    	private String custom;
+
+		CustomEvent(String custom) {
+			super(null, null, null);
+			this.custom = custom;
+		}
+    	
+    }
+    
+    public static class CustomEventFactory implements BaseEventFactory<CustomEvent> {
+
+		@Override
+		public boolean isResponsible(JsonObject messagingEvent) {
+			return messagingEvent.has("custom");
+		}
+
+		@Override
+		public CustomEvent createEventFromJson(JsonObject messagingEvent) {
+			return new CustomEvent(messagingEvent.get("custom").getAsString());
+		}
+    	
+    }
+    
+    @Test
+    public void shouldAllowCustomEventFactory() throws Exception{
+        final Messenger messenger = Messenger.create("test", "test", "test");
+        assertThat(messenger, is(notNullValue()));
+
```

Comment:
Why is it necessary to support such an Event? Is this a Webhook event from facebook for messaging?